examples/nextjs-13/template/apps/host/next.config.js

```diff
@@ -35,6 +36,7 @@ const getRemotes = (/** @type {Boolean} */ isServer) => {
 const nextConfig = {
   cleanDistDir: false,
   webpack: (config, options) => {
+    const gitSHA = execSync(`git rev-list -n 1 HEAD -- .`, { cwd: process.cwd() }).toString().trim();
```

Comment:
gets hash of staged and unstaged files in cwd of build, not hash of whole repo, so its predictable identifier between builds from same source instance. BuildHash stratagy can produce multiple hashes btween client / server. So a source hash of the build context is more reliable to reference between builds of same source code state

Reply:
theoretically I do understand, how this should work
look at screenshot, is it expected behavior?
![image](https://github.com/ZephyrCloudIO/zephyr-examples/assets/1107171/1622adcb-4dd7-4169-9ae3-830b418b35ca)

Reply:
Yeah, @valorkin brought up an important point about using gitHash as a version strategy for regular deployment workflows it will work without much inconvenience, but there are a couple of scenarios where this strategy fails. The biggest one is in the dev environment where things need to be changed often, and your git history doesn't get changed.

Reply:
One suggestions is to use something like [folder-hash](https://www.npmjs.com/package/folder-hash) in a combination with [gitignore-to-glob](https://www.npmjs.com/package/gitignore-to-glob) then parse the ignored files glob and put in the exclusion list to get the current repo fingerprint/hash